### PR TITLE
Add support for issues closed as not planned

### LINKED ISSUE #186: Add support for Issue Closed as Not Planned
image

Issue #4 and #2 of WATcher were Closed as Not Planned but appears as Closed

**Expected Behaviour**
Display the grey symbol like github

### PR DESCRIPTION
**Pull request by joyngjr**
### Summary:

Fixes #186 

Displays issues that are closed as not planned differently from issues which are closed as completed. 

However, this differentiation is not applicable to the filter bar (i.e. filtering closed issues will display issues which are closed as completed and closed as not planned). I'm not sure if we should split 'closed' into 'closed as completed' and 'closed as not planned' in the filter bar as this distinction is only applicable to issues, and may not be compatible with the current design of the filter bar given the increased number of possible invalid combinations (eg. PRs which are closed as not planned, merged issues). Perhaps some refactoring for selection validation in the filter bar would be useful to account for the invalid combinations if more options are added?

#### Type of change:

- ✨ New Feature/ Enhancement
- 🧪 Tests Update

### Changes Made:

- Update graphql library version so schema includes IssueStateReason
- Update queries to retrieve information on stateReason
- Add placeholder `issue-draft` icon as the icon for 'issue closed as not planned' is not in the @primer/octicons library
- Differentiate between issues that are closed as completed and closed as not planned

### Screenshots:

image

### Proposed Commit Message:

```
Differentiates between issues closed as completed and issues closed as not planned.

Previously, all closed issues are displayed in the same way.

Let's display closed issues differently based on its reason for being closed.
```

<details><summary>
<h3>Checklist:</h3>
</summary>

- [ ] I have tested my changes thoroughly.
- [X] I have created tests for any new code files created in this PR or provided a link to a issue/PR that addresses this.
- [X] I have added or modified code comments to improve code readability where necessary.
- [ ] I have updated the project's documentation as necessary.

</details>

### CODE REVIEW

**Review by gycgabriel — COMMENTED** [review]
No worries about the filter at this point, this is just for visual purposes. 

We can use the octicon 'skip'

Do open issues have default stateReason as reopened or is it null?

**Comment by joyngjr** [comment]
@gycgabriel Nope, they do not, so its stateReason is null!

File: src/app/shared/issue-pr-card/issue-pr-card-header/issue-pr-card-header.component.ts [hunk 1]
```diff
@@ -18,14 +18,18 @@ export class IssuePrCardHeaderComponent {
   getOcticon() {
     const type = this.issue.issueOrPr;
     const state = this.issue.state;
+    const stateReason = this.issue.stateReason;
 
     switch (true) {
```
Comment (Eclipse-Dominator) [inline src/app/shared/issue-pr-card/issue-pr-card-header/issue-pr-card-header.component.ts:23]: Might be better to convert this into a regular if-else statement at this point with the increasing number of checks.

**Review by Eclipse-Dominator — CHANGES_REQUESTED** [review]
LGTM except to remove some auto-generated files from graphql

File: src/app/shared/issue-pr-card/issue-pr-card-header/issue-pr-card-header.component.ts (hunk 1 above)
Comment (joyngjr) [reply src/app/shared/issue-pr-card/issue-pr-card-header/issue-pr-card-header.component.ts]: I've made the changes! Though, would it be better to abstract out the checking logic into another service class instead because of the layers of nesting?

File: src/app/core/services/factories/factory.auth.service.ts [hunk 2]
```diff
@@ -30,12 +32,23 @@ export function AuthServiceFactory(
...
   //     userService,
   //     issueService,
+  //     labelService,
   //     phaseService,
   //     githubEventService,
   //     titleService,
   //     logger
   //   );
   // }
   console.log(logger);
```
Comment (Eclipse-Dominator) [inline src/app/core/services/factories/factory.auth.service.ts:42]:
```suggestion
```

not part of this PR but let's remove the debugging statement.

**Review by Eclipse-Dominator — CHANGES_REQUESTED** [review]
LGTM. Let's fix the merge conflict.

**Comment by joyngjr** [comment]
@Eclipse-Dominator @gycgabriel Hi, I've made the changes as described. Could I get your help to review this PR?

**Review by Eclipse-Dominator — APPROVED** [review]
LGTM!
